File: src/test/java/discretemath/bool/operator/ThresholdOperatorTest.java
package discretemath.bool.operator;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

class ThresholdOperatorTest {

	@Test
	void evaluate() {
		ThresholdOperator thresholdOperator = new ThresholdOperator(0.5, new double[]{-1, 1, 2});

		assertFalse(thresholdOperator.evaluate(false, false, false));
		assertTrue(thresholdOperator.evaluate(false, false, true));
		assertTrue(thresholdOperator.evaluate(false, true, false));
		assertTrue(thresholdOperator.evaluate(false, true, true));
		assertFalse(thresholdOperator.evaluate(true, false, false));
		assertTrue(thresholdOperator.evaluate(true, false, true));
		assertFalse(thresholdOperator.evaluate(true, true, false));
		assertTrue(thresholdOperator.evaluate(true, true, true));
	}
}

File: src/test/scala/discretemath/test/bool/operator/ThresholdOperatorTest.scala
package discretemath.test.bool.operator

import discretemath.bool.operator.ThresholdOperator
import org.scalatest.funsuite.AnyFunSuite

class ThresholdOperatorTest extends AnyFunSuite {

  test("evaluate") {
    val thresholdOperator = new ThresholdOperator(0.5, Array[Double](-1, 1, 2))

    assert(!thresholdOperator.%(false, false, false))
    assert(thresholdOperator.%(false, false, true))
    assert(thresholdOperator.%(false, true, false))
    assert(thresholdOperator.%(false, true, true))
    assert(!thresholdOperator.%(true, false, false))
    assert(thresholdOperator.%(true, false, true))
    assert(!thresholdOperator.%(true, true, false))
    assert(thresholdOperator.%(true, true, true))
  }
}